tests/rptest/services/cloud_broker.py

```diff
@@ -32,6 +32,19 @@ def __init__(self, pod, kubectl, logger) -> None:
         self._meta = pod['metadata']
         self.name = self._meta['name']
 
+        spec = pod.get('spec', {})
+        if not spec or 'nodeName' not in spec:
+            health_overview = self.get_health_overview()
```

Comment:
Move this below, to the point of fist use, so if it fails we at least get the logging below which doesn't depend on this.